### PR TITLE
Unit tests for provider/aws

### PR DESCRIPTION
**Pull request by yuriadams**
[task 587](https://github.com/convox/issues-private/issues/587)

### CODE REVIEW

**Review by Twsouza — COMMENTED** [review]
@yuriadams Is there a reason to use the pascal case instead of the snake case for naming the mock files?

**Comment by Twsouza** [comment]
Also, can you look at the `DeepSource: Go`?

File: provider/aws/mocks/CloudFormationAPI.go [hunk 1]
```diff
@@ -0,0 +1,3963 @@
+// Code generated by mockery v2.14.0. DO NOT EDIT.
+
+package mocks
```
Comment (heronrs) [inline provider/aws/mocks/CloudFormationAPI.go:3]: can you put these files under the `pkg/mock` package? maybe `pkg/mock/aws`

**Comment by heronrs** [comment]
All good, a minor ask and DeepSource errors

**Comment by yuriadams** [comment]
> @yuriadams Is there a reason to use the pascal case instead of the snake case for naming the mock files?

@Twsouza , no reason, that was generated by a tool `https://github.com/vektra/mockery`.

**Comment by yuriadams** [comment]
> DeepSource: Go

All good, folks @heronrs @Twsouza

File: pkg/mock/engine.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package mock
```
Comment (heronrs) [inline pkg/mock/engine.go:1]: is this related to k8s?
maybe use pkg/mock/k8s/engine.go?

Comment (yuriadams) [reply pkg/mock/engine.go:1]: @heronrs, is used for both providers. Do you think is it better to move to `pkg/mock/k8s/engine.go` anyway?

Comment (heronrs) [reply pkg/mock/engine.go:1]: oh I see, that's ok then

**Review by heronrs — APPROVED** [review]
